lib/vuln/http_cookie/engine.py

```diff
@@ -0,0 +1,166 @@
+    try:
+        s = conn(target, port, timeout_sec, socks_proxy)
+        if not s:
+            return False
+        else:
+            if target_type(target) != "HTTP" and port == 443:
+                target = 'https://' + target
+            if target_type(target) != "HTTP" and port == 80:
+                target = 'http://' + target
+            req = requests.get(target)
+            value = req.headers['Set-Cookie'].lower()
+            if "samesite" in value:
+                return False
+            else:
+                return True
+
+    except Exception as e:
+        # some error warning
+        return False
+
+
+def __http_cookie(target, port, timeout_sec, log_in_file, language, time_sleep,
+                thread_tmp_filename, socks_proxy, scan_id, scan_cmd):
+    if http_cookie(target, port, timeout_sec, log_in_file, language, time_sleep,
+                 thread_tmp_filename, socks_proxy, scan_id, scan_cmd):
+        info(messages(language, "target_vulnerable").format(target, port,
+                                                            'Samesite cookie flag should be presen inside Set-Cookie header for preventing CSRF attacks'))
+        __log_into_file(thread_tmp_filename, 'w', '0', language)
+        data = json.dumps({'HOST': target, 'USERNAME': '', 'PASSWORD': '', 'PORT': port, 'TYPE': 'http_cookie_vuln',
+                           'DESCRIPTION': messages(language, "vulnerable").format('Samesite cookie flag should be presen inside Set-Cookie header for preventing CSRF attacks'), 'TIME': now(),
+                           'CATEGORY': "vuln",
+                           'SCAN_ID': scan_id, 'SCAN_CMD': scan_cmd})
+        __log_into_file(log_in_file, 'a', data, language)
+        return True
+    else:
+        return False
+
+
+def start(target, users, passwds, ports, timeout_sec, thread_number, num, total, log_in_file, time_sleep, language,
+          verbose_level, socks_proxy, retries, methods_args, scan_id, scan_cmd):  # Main function
+    if target_type(target) != 'SINGLE_IPv4' or target_type(target) != 'DOMAIN' or target_type(target) != 'HTTP':
+        # requirements check
+        new_extra_requirements = extra_requirements_dict()
+        if methods_args is not None:
+            for extra_requirement in extra_requirements_dict():
+                if extra_requirement in methods_args:
+                    new_extra_requirements[
+                        extra_requirement] = methods_args[extra_requirement]
+        extra_requirements = new_extra_requirements
+        if ports is None:
+            ports = extra_requirements["http_cookie_vuln_ports"]
+        if target_type(target) == 'HTTP':
+            target = target_to_host(target)
+        threads = []
+        total_req = len(ports)
+        thread_tmp_filename = '{}/tmp/thread_tmp_'.format(load_file_path()) + ''.join(
+            random.choice(string.ascii_letters + string.digits) for _ in range(20))
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Standard pseudo-random generators are not suitable for security/cryptographic purposes.](https://app.codacy.com/manual/zdresearch/OWASP-Nettacker/pullRequest?prid=5181770)